src/utils/date_utils.py

```diff
@@ -20,6 +20,22 @@ def get_years_difference_from_current_date(utc_hour_offset: str, date: str, date
 def is_current_date(date: str, date_format: str = '%Y-%m-%d') -> bool:
     if not date:
         return False
+
+    if not is_valid_format_date(date, date_format):
+        return False
+
     current_day_month = get_current_day_month(EnvManager.UTC_HOUR_OFFSET)
     date = datetime.strptime(date, date_format)
     return current_day_month == (date.day, date.month)
+
+
+def is_valid_format_date(date: str, date_format: str) -> bool:
+    if not date:
+        return False
+
+    try:
+        datetime.strptime(date, date_format)
```

Comment:
Maybe try to validate this in a different way `strptime` is kind of expensive and if we are not doing useful things with it is better to not overuse the function